--- lin_tape-3.0.33/lower.c.orig	2018-10-27 12:12:06.000000000 +0200
+++ lin_tape-3.0.33/lower.c	2018-10-27 14:11:03.207396481 +0200
@@ -276,8 +276,13 @@
 			pfo_cmd->bufflen, local_raw_sense, pfo_cmd->timeout,
 			pfo_cmd->retries, pfo_cmd->flags, &pfo_cmd->resid_len);
 #else
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,19,0)
 			pfo_cmd->bufflen, local_raw_sense, &local_sense_data, pfo_cmd->timeout,
 			pfo_cmd->retries, pfo_cmd->flags, 0, &pfo_cmd->resid_len);
+#else
+			pfo_cmd->bufflen, NULL, &local_sense_data, pfo_cmd->timeout,
+			pfo_cmd->retries, pfo_cmd->flags, 0, &pfo_cmd->resid_len);
+#endif
 #endif
 	} else {
 		/* SCSI Command from user space */
@@ -291,8 +296,12 @@
 	scsi_device_put(sdev);
 	unlock_apfo_path(path_num);
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,19,0)
 	valid_sense = scsi_normalize_sense(local_raw_sense,
 				SCSI_SENSE_BUFFERSIZE, &local_sense_data);
+#else
+	valid_sense = scsi_sense_valid(&local_sense_data);
+#endif
 	/* Put the SCSI command results in the structure */
 	pfo_cmd->errors = result;
 	pfo_cmd->scsi_status_code = result & 0xff;
