--- lin_tape-3.0.33/lin_tape_scsi_tape.c~	2018-12-28 14:09:16.000000000 +0100
+++ lin_tape-3.0.33/lin_tape_scsi_tape.c	2018-12-28 14:12:49.237558696 +0100
@@ -11843,6 +11843,9 @@
 #elif LINUX_VERSION_CODE >= KERNEL_VERSION(4,6,0)
 	mapped_pages = get_user_pages((ulong)user_location, number_pages,
 			                      rw, 0, pages, NULL);
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4,4,169)
+	mapped_pages = get_user_pages(current, current->mm,
+		(ulong)user_location, number_pages, rw ? FOLL_WRITE : 0, pages, NULL);
 #else
 	mapped_pages = get_user_pages(current, current->mm,
 		(ulong)user_location, number_pages, rw, 0, pages, NULL);
