--- lin_tape-3.0.20/lin_tape_ioctl_tape.c~	2017-09-16 10:27:48.000000000 +0200
+++ lin_tape-3.0.20/lin_tape_ioctl_tape.c	2017-09-16 10:31:03.642711575 +0200
@@ -6465,7 +6465,11 @@
 
 	tmp_rq = (void*)sdev->request_queue;
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4,11,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,13,0)
+	scsi_req_init(scsi_req(req));
+#else
 	scsi_req_init(req);
+#endif
 #else
 	req->cmd_type = REQ_TYPE_BLOCK_PC;
 #endif
--- lin_tape-3.0.20/lower.c~	2017-09-16 10:27:48.000000000 +0200
+++ lin_tape-3.0.20/lower.c	2017-09-16 10:32:08.377461388 +0200
@@ -470,7 +470,11 @@
 	}
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4,11,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,13,0)
+	scsi_req_init(scsi_req(rq));
+#else
 	scsi_req_init(rq);
+#endif
 #else
 	rq->cmd_type = REQ_TYPE_BLOCK_PC;
 #endif
