--- lin_tape-3.0.71/Makefile.orig	2025-12-15 00:50:47.000000000 +0100
+++ lin_tape-3.0.71/Makefile	2025-12-15 00:56:52.103317958 +0100
@@ -19,75 +19,12 @@
 
 lin_tape-objs := join.o lin_tape_scsi_config.o lin_tape_scsi_tape.o lin_tape_scsi_trace.o \
              lin_tape_ioctl_tape.o lin_tape_ioctl_changer.o lin_tape_extra_ioctl.o
-RHELRELEASE := $(shell [ -f "/etc/redhat-release" ] && echo 1 || echo 0)
-ifeq ($(RHELRELEASE), 1)
-KERNELDIR ?= /lib/modules/$(shell uname -r)/build
-KERNELVER = $(shell uname -r 2>/dev/null|sed "s/[\.-]/_/g"|sed "s/\([0-9]*_[0-9]*_[0-9]*_[0-9]*\).*/\1/")
-EXTRA_CFLAGS += -DKERNELVERSION_${KERNELVER}
-RHEL94 := $(shell grep -qi 9.4 /etc/*release && echo 1 || echo 0)
-ifeq ($(RHEL94), 1)
-EXTRA_CFLAGS += -DRHEL94
-else
-RHEL95 := $(shell grep -qi 9.5 /etc/*release && echo 1 || echo 0)
-ifeq ($(RHEL95), 1)
-EXTRA_CFLAGS += -DRHEL94
-EXTRA_CFLAGS += -DRHEL95
-else
-RHEL96 := $(shell grep -qi 9.6 /etc/*release && echo 1 || echo 0)
-ifeq ($(RHEL96), 1)
-EXTRA_CFLAGS += -DRHEL94
-EXTRA_CFLAGS += -DRHEL95
-EXTRA_CFLAGS += -DRHEL96
-endif
-endif
-endif
-else
-UBUNTURELEASE := $(shell [ -f "/etc/os-release" ] && grep -qi Ubuntu /etc/os-release && echo 1 || echo 0)
-ifeq ($(UBUNTURELEASE), 1)
-EXTRA_CFLAGS += -DUBUNTUVERSION
-else
-EXTRA_CFLAGS += -DSLESVERSION
-KERNELDIR ?= /lib/modules/$(shell uname -r)/build
-KERNELVER = $(shell uname -r 2>/dev/null|sed "s/[\.-]/_/g"|sed "s/\([0-9]*_[0-9]*_[0-9]*_[0-9]*\).*/\1/")
-EXTRA_CFLAGS += -DSLESKERNEL_${KERNELVER}
-SLESSP2 := $(shell grep -qi SP2 /etc/*release && echo 1 || echo 0)
-SLESSP5 := $(shell grep -qi SP5 /etc/*release && echo 1 || echo 0)
-SLES15SP6 := $(shell grep -qi "15 SP6" /etc/*release && echo 1 || echo 0)
-ifeq ($(SLESSP2), 1)
-EXTRA_CFLAGS += -DSLESSP2
-endif
-ifeq ($(SLESSP5), 1)
-EXTRA_CFLAGS += -DSLESSP5
-endif
-ifeq ($(SLES15SP6), 1)
-EXTRA_CFLAGS += -DSLES15SP6
-endif
-endif
-endif
 
 ifeq ($(SFMP), 1)
 EXTRA_CFLAGS += -DSFMPENABLED
 endif
 
 PROC := $(shell uname -m)
-ifeq ($(UBUNTURELEASE), 0)
-ifeq ($(PROC), ppc64)
-EXTRA_CFLAGS += -m64
-endif
-
-ifeq ($(PROC), powerpc64)
-EXTRA_CFLAGS += -m64
-endif
-
-ifeq ($(PROC), powerpc)
-EXTRA_CFLAGS += -m64 -msoft-float
-LDFLAGS += -melf64ppc
-endif
-
-ifeq ($(PROC), x86_64)
-EXTRA_CFLAGS += -mcmodel=kernel
-endif
-endif
 
 VERSION := $(word 1,$(VERSIONRELEASE))
 RELEASE := $(word 2,$(VERSIONRELEASE))
--- lin_tape-3.0.71/upper.c~	2025-06-30 16:30:30.000000000 +0200
+++ lin_tape-3.0.71/upper.c	2025-12-15 01:03:04.216651273 +0100
@@ -312,9 +312,9 @@
 
 #ifdef KERNELVERSION_5_14_0_70 
 	if (blk_verify_command(pfo_cmd->scsi_cmd, mode & FMODE_WRITE))
-#elif defined (UBUNTUVERSION) | LINUX_VERSION_CODE >= KERNEL_VERSION(6,0,0) || (LINUX_VERSION_CODE >= KERNEL_VERSION(5,14,0) && (LINUX_VERSION_CODE < KERNEL_VERSION(5, 14, 1) || defined(SLESSP5)))
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(5,14,0)
 	if (!scsi_cmd_allowed(pfo_cmd->scsi_cmd, mode & FMODE_WRITE))
-#elif LINUX_VERSION_CODE > KERNEL_VERSION(3,12,0) && (defined(SLESVERSION) || defined(UBUNTUVERSION))
+#elif LINUX_VERSION_CODE > KERNEL_VERSION(3,12,0)
 	if (blk_verify_command(pfo_cmd->scsi_cmd, mode & FMODE_WRITE))
 #else
 	/* RHEL 6.4 and beyond */
