--- lin_tape-1.74.0/Makefile~	2012-08-21 19:32:26.000000000 +0200
+++ lin_tape-1.74.0/Makefile	2012-09-29 13:26:24.816711524 +0200
@@ -5,7 +5,7 @@
 #include $(PWD)/Kbuild
 
 else
-VERSIONRELEASE := $(shell sed --regexp-extended -e '{/\#define LIN_TAPE_VERSION/ s/.*"([0-9.]*)-?([A-Za-z0-9.]*)"/\1 \2/p;d}' lin_tape_tape.h)
+VERSIONRELEASE := $(shell sed --regexp-extended -e '{/\#define LIN_TAPE_VERSION/ s/.*"([0-9.]*)-?([A-Za-z0-9.]*)"/\1 \2/p;d}' $(M)/lin_tape_tape.h)
 endif
 
 obj-m := stmp.o
