diff -urNp -x '*.orig' lin_tape-3.0.52.org/Makefile lin_tape-3.0.52/Makefile
--- lin_tape-3.0.52.org/Makefile	2021-02-24 22:06:04.385259650 +0100
+++ lin_tape-3.0.52/Makefile	2021-02-24 22:06:04.421926205 +0100
@@ -5,7 +5,7 @@ ifneq ($(KERNELRELEASE),)
 #include $(PWD)/Kbuild
 
 else
-VERSIONRELEASE := $(shell sed --regexp-extended -e '{/\#define LIN_TAPE_VERSION/ s/.*"([0-9.]*)-?([A-Za-z0-9.]*)"/\1 \2/p;d}' lin_tape_tape.h)
+VERSIONRELEASE := $(shell sed --regexp-extended -e '{/\#define LIN_TAPE_VERSION/ s/.*"([0-9.]*)-?([A-Za-z0-9.]*)"/\1 \2/p;d}' $(M)/lin_tape_tape.h)
 endif
 
 ifeq ($(SFMP), 1)
